plugins/scripts/upload/trns_transform_mzML_LCMS_to_MetaboliteAtlas2_MAFileInfo.py

```diff
@@ -0,0 +1,237 @@
+    """
+    Converts mzML file to MetaboliteAtlas2_MAFileInfo json string.
+
+    Args:
+        shock_service_url: A url for the KBase SHOCK service.
+        handle_service_url: A url for the KBase Handle Service.
+        output_file_name: A file name where the output JSON string should be
+                          stored.
+                          If the output file name is not specified the name
+                          will default
+                          to the name of the input file appended with
+                           '_finfo'.
+        input_directory: The directory where files will be read from.
+        working_directory: The directory the resulting json file will be
+                           written to.
+        shock_id: Shock id for the hdf file if it already exists in shock
+        handle_id: Handle id for the hdf file if it already exists as a
+                    handle
+        input_mapping: JSON string mapping of input files to expected types.
+                       If you don't get this you need to scan the input
+                       directory and look for your files.
+        level: Logging level, defaults to logging.INFO.
+        atlases: List of MetaboliteAtlas atlas IDs.
+        mzml_file_name: Name of the file, optional.  Defaults to the file name.
+        polarity: Run polarity.
+        group: Run group.
+        inclusion_order: Run inclusion_order.
+        retention_correction: Run retention_correction.
+        normalization_factor: Run normalization factor.
+
+    Returns:
+        JSON files on disk that can be saved as a KBase workspace objects.
+
+    Authors:
+        Steven Silvester
+    """
+
+    if logger is None:
+        logger = script_utils.stderrlogger(__file__)
+
+    logger.info("Starting conversion of mzML to MetaboliteAtlas2.MAFileInfo")
+    token = os.environ.get('KB_AUTH_TOKEN')
+
+    if not working_directory or not os.path.isdir(working_directory):
+        raise Exception("The working directory {0} is not a valid directory!"
+                        .format(working_directory))
+
+    logger.info("Scanning for mzML files.")
+
+    valid_extensions = [".mzML"]
+
+    files = os.listdir(input_directory)
+    mzml_files = [x for x in files
+                             if os.path.splitext(x)[-1] in valid_extensions]
+
+    assert len(mzml_files) != 0
+
+    logger.info("Found {0} files".format(len(mzml_files)))
+
+    for fname in mzml_files:
+        path = os.path.join(input_directory, fname)
+
+        if not os.path.isfile(path):
+            raise Exception("The input file name {0} is not a file!"
+                            .format(path))
+
+        hdf_file = mzml_loader.mzml_to_hdf(path)
+
+        if shock_service_url:
+            shock_info = script_utils.upload_file_to_shock(logger,
+                    shock_service_url, hdf_file, token=token)
+
+        run_info = dict()
+        run_info['name'] = name or fname.replace('.mzML', '')
```

Comment:
And one more lost ```name``` ref: https://github.com/kbase/transform/pull/151/files#diff-a6e65713be5cb812af79d41b4d89fef5R112